### PR TITLE
fix: tx SDK changes and Substrate 2.0 update

### PR DESCRIPTION
**Pull request by Dudleyneedham**
## fixes https://github.com/KILTprotocol/ticket/issues/758

Changes to the types and the way we handle the DIDs.
Update to substrate 2.0

## How to test:

Yarn start and go through the workflow.

## Checklist:

- [ ] I have verified that the code works
- [ ] I have verified that the code is easy to understand
  - [ ] If not, I have left a well-balanced amount of inline comments
- [ ] I have [left the code in a better state](https://deviq.com/boy-scout-rule/)
- [ ] I have documented the changes (where applicable)

### CODE REVIEW

File: src/services/DidService.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ class DidService {
     } as IContact)
 
     const status = await did.store(myIdentity.identity)
-    if (status.isError) {
+    if (status.isEmpty) {
```
Comment (Dudleyneedham) [inline src/services/DidService.ts:37]: With the changes, it seems that we cannot have the `isError` not sure what the equivalent is. 

Will check more about the types to find a better solution.

**Comment by Dudleyneedham** [comment]
I think a question for this PR and the general way we handle transactions on the demo-client. Should we stop the user from making more than one transaction like we are currently doing or should this run in the background only notifying when submitted and finalised, respectively?

**Comment by tjwelde** [comment]
In the future, we would want the transactions to be run in the background, but for that, we need the multi-txs.
An intermediate goal could be, that we wait for "IN_BLOCK" and not for "FINALIZED".
But both are out of scope for this PR

**Review by tjwelde — APPROVED** [review]
LGTM

**Pull request merged**